### PR TITLE
Remove cgi.FieldStorage (based on #769)

### PR DESCRIPTION
**Pull request by scottbarnes**
I took the changes from #769 and modified them as little as possible to get Open Library working with it. See internetarchive/openlibrary#8313.

This still needs more testing.

@tfmorris
@cclauss 
@mekarpeles

### CODE REVIEW

File: web/webapi.py [hunk 1]
```diff
@@ -423,47 +406,60 @@ def rawinput(method=None):
...
+                    except IndexError:
+                        a = {}
+
             else:
-                d = data()
-                if isinstance(d, str):
-                    d = d.encode("utf-8")
-                fp = BytesIO(d)
-                a = cgiFieldStorage(fp=fp, environ=e, keep_blank_values=1)
+                post_data = data().decode("utf-8")
```
Comment (mekarpeles) [inline web/webapi.py:439]: Just making sure we're not accidentally losing anything in this `decode` (or hitting any other unexpected results)

File: web/webapi.py [hunk 2]
```diff
@@ -423,47 +406,60 @@ def rawinput(method=None):
...
-        b = dictify(cgiFieldStorage(environ=e, keep_blank_values=1))
-
-    def process_fieldstorage(fs):
-        if isinstance(fs, list):
-            return [process_fieldstorage(x) for x in fs]
-        elif fs.filename is None:
-            return fs.value
+        env["REQUEST_METHOD"] = "GET"
+        b = dict(
+            urllib.parse.parse_qs(env.get("QUERY_STRING", ""), keep_blank_values=True)
```
Comment (mekarpeles) [inline web/webapi.py:449]: NB: QUERY_STRING previously handled within `cgiFieldStorage` constructor.

**Comment by mekarpeles** [comment]
@anandology + @cclauss + @tfmorris any feedback or 👍 👎 ?

Who has the final say for webpy merging? I'm not sure how it's being PM'd right now, cc: @cclauss

**Review by cclauss — APPROVED** [review]
LGTM.  Thanks, @tfmorris, @scottbarnes, @mekarpeles for the collaboration.

Monday Python 3.12 should ship.
One year from Monday Python 3.12 should ship and web.py should be ready.  ;-)

**Review by tfmorris — COMMENTED** [review]
This had been merged by the time I saw the ping requesting review, so I didn't bother, but circling back to do a post-hoc review now since there was an issue with it.

**Comment by cclauss** [comment]
https://github.com/webpy/webpy/issues/776

**Pull request merged**